applications/serial_lte_modem/src/main.c

```diff
@@ -338,32 +339,73 @@ static void check_app_fota_status(void)
 int lte_auto_connect(void)
 {
 	int err = 0;
-
-#if defined(CONFIG_SLM_CARRIER)
+#if defined(CONFIG_SLM_AUTO_CONNECT)
 	int stat;
+	struct network_config {
+		/* Network-specific default system mode configuration by %XSYSTEMMODE */
+		int lte_m_support;     /* 0 ~ 1 */
+		int nb_iot_support;    /* 0 ~ 1 */
+		int gnss_support;      /* 0 ~ 1 */
+		int lte_preference;    /* 0 ~ 4 */
+		/* Network-specific default PDN configuration by +CGDCONT and +CGAUTH */
```

Comment:
Split based on AT-command?

Reply:
Let's leave it as it is as they are all part of PDN configuration

Reply:
Fair enough.